src/main/java/org/sonarlint/intellij/actions/MarkAsResolvedAction.kt

```diff
@@ -0,0 +1,263 @@
+                    addComment(project, module, issueKey, comment)
+                }
+                .exceptionally { error ->
+                    SonarLintConsole.get(project).error("Error while marking issue as resolved", error)
+
+                    notifyError(project, "<b>SonarLint - Unable to mark as resolved</b>", "Could not mark the issue as resolved.")
+                    null
+                }
+        }
+
+        private fun notifyError(project: Project, title: String, content: String) {
+            val notification = GROUP.createNotification(
+                title,
+                content,
+                NotificationType.ERROR
+            )
+            notification.isImportant = true
+            notification.notify(project)
+        }
+
+        private fun updateUI(
+            project: Project,
+            isTaintVulnerability: Boolean,
+            localTaintVulnerability: LocalTaintVulnerability?,
+            liveIssue: LiveIssue?,
+        ) {
+            runOnUiThread(project) {
+                val toolWindowService = getService(project, SonarLintToolWindow::class.java)
+
+                if (isTaintVulnerability) {
+                    toolWindowService.markAsResolved(localTaintVulnerability)
+                } else {
+                    toolWindowService.markAsResolved(liveIssue)
+                }
+                getService(project, CodeAnalyzerRestarter::class.java).refreshOpenFiles()
+            }
+        }
+
+        private fun addComment(project: Project, module: Module, issueKey: String, comment: String) {
+            getService(BackendService::class.java)
+                .addCommentOnIssue(module, issueKey, comment)
+                .exceptionally { error ->
+                    SonarLintConsole.get(project).error("Error while adding a comment on the issue", error)
+                    notifyError(project, "<b>SonarLint - Unable to add a comment</b>", "Could not add a comment on the issue.")
+                    null
+                }
+        }
+
+        private fun checkPermission(project: Project, connection: ServerConnection, issueKey: String): CheckStatusChangePermittedResponse? {
+            val checkTask = CheckIssueStatusChangePermission(project, connection, issueKey)
+            return try {
+                ProgressManager.getInstance().run(checkTask)
+            } catch (e: Exception) {
+                SonarLintConsole.get(project).error("Error while retrieving the list of allowed statuses for issues", e)
+                displayErrorNotification(project, errorTitle, "Could not check status change permission", GROUP)
+                null
+            }
+        }
+
+        private fun confirm(project: Project, productName: String, issueStatus: IssueStatus): Boolean {
+            if (PropertiesComponent.getInstance().getBoolean(SKIP_CONFIRM_DIALOG_PROPERTY, false)) {
+                return true
+            }
+            return MessageDialogBuilder.okCancel(
```

Comment:
Small detail but we can use Kotlin `return if () { } else { }` to avoid having 2 returns